packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -941,35 +960,74 @@ export class GarbageCollector implements IGarbageCollector {
             };
         }
 
-        const newSerializedSummaryState = JSON.stringify(generateSortedGCState(gcState));
+        const serializedGCState = JSON.stringify(generateSortedGCState(gcState));
+        const serializedTombstones = this.tombstones.length > 0 ? JSON.stringify(this.tombstones.sort()) : undefined;
```

Comment:
Why the different treatment of `serializedGCState` vs `serializedTombstones`? It means there's an extra if statement check

Reply:
It's just an optimization. We always write GC state because GC is enabled by default so more often than not there will be GC data to write.
For tombstones, there is a chance that there is no data because either tombstoning is disabled or even if enabled, chances of tombstones being there is less. So, in those cases, we won't write the tombstone blob at all.

Reply:
I was talking with @markfields and he was thinking that we would always store whether or not a blob was tombstoned in the summary. Or am I misunderstanding something?

Reply:
I am not sure I follow. Basically, the behavior is that if something is tombstoned, we write it into the tombstone blob. If nothing is tombstoned, we don't write the tombstone blob at all.

Currently, if tombstoning is not enabled via a feature flag, we don't write tombstones at all. This will change in the future where we will start doing this by default.

Reply:
Hmmm ok, I'm good with that